PSSwagger/SwaggerUtils.psm1

```diff
@@ -1954,41 +1933,97 @@ function Get-Response
         OutputTypeBlock = $OutputTypeBlock
     }
 }
-function Get-CSharpModelName
-{
+function Get-CSharpModelName {
     param(
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string]
-        $Name
+        $Name,
+
+        [Parameter(Mandatory = $false)]
+        [switch]
+        $ReplaceBasicLatinCharacters
     )
 
     # Below logic is as per AutoRest
-    $Name = $Name -replace '[[]]','Sequence'
+    $Name = $Name -replace '[[]]', 'Sequence'
+    if ($ReplaceBasicLatinCharacters) {
+        $chars = $Name.ToCharArray()
+        $Name = ""
+        foreach ($char in $chars) {
+            # Not sure how to box this properly
+            if ($script:AutoRestLaticCharacters.ContainsKey($char+"")) {
+                $Name += $script:AutoRestLaticCharacters[$char+""]
+            } else {
+                $Name += $char
+            }
+        }
+    }
+
     # Remove special characters
-    $Name = $Name -replace '[^a-zA-Z0-9_-]',''
+    $Name = $Name -replace '[^a-zA-Z0-9_-]', ''
 
     # AutoRest appends 'Model' to the definition name when it is a C# reserved word.
-    if($script:CSharpReservedWords -contains $Name) {        
+    if ($script:CSharpReservedWords -contains $Name) {        
         $Name += 'Model'
     }
 
     return Get-PascalCasedString -Name $Name
 }
 
+function Get-CSharpMethodName {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]
+        $Name
+    )
+
+    # Eventually load the AutoRest dll - for now this doesn't work without the dotnet CLI available, so have to hack around it for now
+    # Including the code to get the AutoRest.CSharp dll location
+    <#
+    $command = Get-Command autorest
+	$path = ""
+    if ($command) {
+        $output = & autorest --version
+        foreach ($line in $output) {
+            if ($line.Contains("@microsoft.azure/autorest.csharp")) {
+                $index = $line.Length - 1
+                while (-not $path -or -not (Test-Path -Path $path)) {
+                    $path = $line[$index] + $path
+                    $index -= 1
+                }
+
+                break
+            }
+        }
+		
+		if ($path) {
+			$path = (Get-ChildItem -Path "$path$([System.IO.Path]::DirectorySeparatorChar)autorest.csharp.dll" -File -Recurse | Select-Object -First 1).FullName
+		} else {
+			$path = Join-Path -Path $env:UserProfile -ChildPath ".autorest" | Join-Path -ChildPath "plugins" | Join-Path -ChildPath "autorest"
+			$versions = @()
+			foreach ($item in Get-ChildItem $path -Directory) {
+				$versions += Split-Path -Path $item.FullName -Leaf
+			}
+			$path = Join-Path -Path $path -ChildPath ($versions | Sort-Object -Descending | Select-Object -First 1) | Join-Path -ChildPath "autorest.csharp.dll"
+		}
+    }#>
```

Comment:
Minor: consider removing the commented code for getting the autorest.csharp.dll path.